[11.0][ADD] website_media_size

tarteo opened the pull request:
This module shows the size of the media in the media selector (only for media that was uploaded).

![screenshot](https://user-images.githubusercontent.com/10028499/36030395-554177c6-0da7-11e8-8054-e7702bbb72d7.png)

simahawk commented on website_media_size/static/src/js/website_media_size.js:
```diff
@@ -0,0 +1,42 @@
+            var res = this._super.apply(this, arguments);
+            this.include_file_size(this.imageDialog);
+            this.include_file_size(this.documentDialog);
+             // FIXME
```
?

tarteo replied:
The this.search(); is a weird fix to update the size values. I couldn't find a proper solution. There must be a better way to fix it.
In the start of super it already executes .search(). I can probably just remove the FIXME because I don't think it's actually an issue.

yajo commented on website_media_size/static/src/css/website_media_size.css:
```diff
@@ -0,0 +1,12 @@
+.o_existing_attachment_cell .o_existing_attachment_size {
+    position: absolute;
+    padding: 3px;
+    background-color: rgba(0, 0, 0, 0.8);
```
You can use "opacity"

tarteo replied:
I need the background to be black.

yajo replied:
... Only if the foreground is white :stuck_out_tongue_winking_eye:

tarteo replied:
A black background will create contrast so it will be easy to read. Opacity will also apply it on the text making it less readable in all cases.

yajo commented on website_media_size/static/src/css/website_media_size.css:
```diff
@@ -0,0 +1,12 @@
+    color: #FFF;
+    border-radius: 0 0 0 2px;
+    cursor: pointer;
+    font-size: 80%;
```
It's way better if you use a less file and use Bootstrap variables for such properties, or even better to use a normal bootstrap class such as: https://getbootstrap.com/docs/3.3/components/#labels

tarteo replied:
Yes, but I would be changing the label's style to match the other buttons.

yajo replied:
Well, the point is that if you leverage Bootstrap, you make your thing look good no matter the theme; if you don't, i.e. a dark theme would render this ugly.

Non-blocking in any case IMHO

tarteo replied:
Yes, I understand it.

Yajo commented on website_media_size/static/src/js/website_media_size.js:
```diff
@@ -0,0 +1,41 @@
+ * License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). */
+
+odoo.define('website_media_size', function(require) {
+    var MediaDialog = require('web_editor.widget').MediaDialog;
```
"use strict"

yajo commented on website_media_size/static/src/js/website_media_size.js:
```diff
@@ -0,0 +1,41 @@
+        start: function() {
+            var res = this._super.apply(this, arguments);
+            this.include_file_size(this.imageDialog);
+            this.include_file_size(this.documentDialog);
```
Wouldn't it be better (less monkey-y 🙈) to patch each of these 2 classes' `_rpc` method and add the required field to `arguments[0].kwargs.fields` if the model is `ir.attachment`?

tarteo replied:
That's a great idea! Why I didn't taught of that. That's genius. :smile: I'll apply it asap.

tarteo commented:
@Yajo I applied your suggestion on Widget class because ImageDialog is not accessible. I do a check on template so it is fine.
I also replaced the screenshot because of the possibly copyrighted picture of the monkey. I replaced it with a non copyrighted picture of a monkey.

yajo commented on website_media_size/static/src/js/website_media_size.js:
```diff
@@ -2,40 +2,32 @@
  * License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). */
 
 odoo.define('website_media_size', function(require) {
+"use strict";
```
Indent this line please :innocent:

yajo commented on website_media_size/static/src/js/website_media_size.js:
```diff
@@ -2,40 +2,32 @@
+        }
+    });
+
+    // ImageDialog of the web_editor.widget module is not available
```
I took the freedom to open https://github.com/odoo/odoo/pull/23180, I'm pretty sure our friend @ged-odoo will merge that soon. :wink:

tarteo replied:
Thanks!!

Fuchur777 commented:
@simahawk @Yajo does it now LGTY? Can you amend your review?

simahawk commented on website_media_size/static/src/js/website_media_size.js:
```diff
@@ -0,0 +1,33 @@
+        }
+    });
+
+    // ImageDialog of the web_editor.widget module is not available
```
@tarteo https://github.com/odoo/odoo/pull/23180 got merged. Do you still need this?

tarteo replied:
No, had no time to change it yet.

yajo commented on website_media_size/static/src/js/website_media_size.js:
```diff
@@ -0,0 +1,31 @@
+        ),
+        start: function() {
+            var res = this._super.apply(this, arguments);
+            this.search();
```
Hmm why this?

Also why patching different methods on different classes? I'm not saying it's wrong, just weird at 1st sight... :thinking:

tarteo replied:
You're right this is not needed anymore. I removed it and moved the xml dependency to ImageDialog.
